packages/lit-html/src/directive.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {
+  _$private,
+  AttributePart,
+  DirectiveClass,
+  DirectiveParameters,
+  DirectiveResult,
+  Disconnectable,
+  NodePart,
+  PartInfo,
+  Part,
+} from './lit-html';
+export {DirectiveClass, DirectiveParameters, DirectiveResult} from './lit-html';
+
+const resolveDirective = _$private._resolveDirective;
+
+export const ATTRIBUTE_PART = 1;
+export const NODE_PART = 2;
+export const PROPERTY_PART = 3;
+export const BOOLEAN_ATTRIBUTE_PART = 4;
+export const EVENT_PART = 5;
+
+/**
+ * Creates a user-facing directive function from a Directive class. This
+ * function has the same parameters as the directive's render() method.
+ *
+ * WARNING: The directive and part API changes are in progress and subject to
+ * change in future pre-releases.
+ */
+export const directive = <C extends DirectiveClass>(c: C) => (
+  ...values: DirectiveParameters<InstanceType<C>>
+): DirectiveResult<C> => ({
+  _$litDirective$: c,
+  values,
+});
+
+/**
+ * Base class for creating custom directives. Users should extend this class,
+ * implement `render` and/or `update`, and then pass their subclass to
+ * `directive`.
+ *
+ * WARNING: The directive and part API changes are in progress and subject to
```

Comment:
Should we go ahead and remove this, so we don't forget?

Reply:
done

Reply:
done